modules/mocked_authentication/lib/mockdata/reader.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module MockedAuthentication
+  module Mockdata
+    class Reader
+      def self.find_credentials(credential_type:)
+        file_names = get_file_names(credential_type)
+        get_mocked_data(file_names)
+      end
+
+      class << self
+        private
+
+        def get_file_names(type)
+          file_directory = get_file_directory_for_type(type)
+          users = Dir.entries(file_directory).select { |file| validate_file_name(file) ? file : nil }
```

Comment:
Ah yeah, this is actually a good error, we do need to fix this. What we have to do is just separate the input from being interpolated into a file path directly, I'd do something like this:

> case type
  when SignIn::Constants::Auth::IDME
    IDME_FILEPATH
  when SignIn::Constants::Auth::LOGINGOV
    LOGINGOV_FILEPATH
    ...

Reply:
And `IDME_FILEPATH` and LOGINGOV_FILEPATH` would just be `idme` and `logingov`, which seems silly, but it makes it so that a user cannot put in arbitrary text and have us pull a file (I know the controller blocks this but still good to block it here too)